### ISSUE TITLE
ESBMC_unreachable does not guarantee UNKNOWN verification result

### ISSUE THREAD

**Issue by emanino** [esbmc/esbmc#1511]
I am working with an "empty" implementation of the expf() function, based on the __ESBMC_unreachable() intrinsic. Unfortunately, the code below returns VERIFICATION SUCCESS despite calling expf() multiple times. Note that with a "proper" implementaion of expf() the code below should return a counterexample for any value of the nondeterministic x[0]. Indeed the softmax function is monotonic, so y[2] <= y[3] is always true.

I run ESBMC with the following flags: `./esbmc --no-div-by-zero-check --force-malloc-success   --no-align-check  --floatbv   --no-vla-size-check ./../../sv-benchmarks/c/neural-networks/softsign_w4_r1_case_0_unsafe.c-amalgamation.i --32 --witness-output witness.graphml  --no-pointer-check  --no-bounds-check --error-label ERROR   --k-induction`

Are we experiencing another strange intereaction with k-induction?

```
#include <math.h>

float expf(float x)
{
__ESBMC_HIDE:;
    
	// unimplemented operational models
    // should return VERIFICATION UNKNOWN
    __ESBMC_unreachable();

    return 0.0f;
}

void softmax(const float* x, float* y, int n)
{
	int i;
	
	/* find max input value to avoid overflow */
	float max = x[0];
	for(i = 1; i < n; ++i)
		if(x[i] > max)
			max = x[i];
	
	/* compute the element-wise exponential while keeping track of their sum */
	float sum = 0.0f;
	for(i = 0; i < n; ++i) {
		float z = expf(x[i] - max);
		sum += z;
		y[i] = z;
	}
	
	/* divide by the sum, i.e. make the result sum up to one */
	for(i = 0; i < n; ++i)
		y[i] /= sum;
}

#define SOFTMAX_CHECK_NEXT 1e-5f

int main() /* check_non_decreasing_size_4 */
{
	float x[4], y[4];
	
	x[0] = nondet_float();
	x[1] = x[0] + SOFTMAX_CHECK_NEXT;
	x[2] = 0.123456f;
	x[3] = 0.654321f;
	
	__ESBMC_assume(!isnan(x[0]) && !isinf(x[0]));
	
	softmax(x, y, 4);
	
	__ESBMC_assert(isgreaterequal(y[2], y[3])); /* Expected result: verification failure */

    return 0;
}
```

**Comment by fbrausse** [comment]
You're right, just adding the assertion isn't going to work. We need to crash esbmc (or at least make it output something the wrapper interprets as Unknown). I tried that with this intrinsic, but as it's just another regular assertion it's susceptible to bugs and interplay with other assertions. I'd image this should work: `if (!guard.is_false()) abort()` instead of the current implementation of the intrinsic.
